web/src/condense.ts

```diff
@@ -71,17 +80,27 @@ export function uncollapse($row) {
     };
 
     // We also need to collapse this message in the home view
-    const $home_row = message_lists.home.get_row(rows.id($row));
+    const $home_row = message_lists.home.get_row(id);
 
     process_row($row);
     process_row($home_row);
 }
 
-export function collapse($row) {
+export function collapse($row: JQuery): void {
     assert(message_lists.current !== undefined);
+    assert(message_lists.home !== undefined);
     // Collapse a message, hiding the condensed message [More] or
     // [Show less] link if necessary.
-    const message = message_lists.current.get(rows.id($row));
+    const id = rows.id($row)
+    if (id === undefined) {
+        return;
+    }
+
+    const message = message_lists.current.get(id);
+    if (message === undefined) {
```

Comment:
> There is no need to check if the message is undefined or not here is why.
we are checking if we are doing assertation above (line 90) to see if current is undefined or not.
which means, if message_lists exists then there exists a valid id 

here to

Reply:
@Mahhheshh thanks for the review. I did see the assertation. However, Typescript continue to show this error when you hover on `message`: message may be undefined. Should I ignore this error?